pyomo/contrib/solver/solvers/ipopt.py

```diff
@@ -431,16 +448,26 @@ def solve(self, model, **kwds):
                     results.termination_condition = TerminationCondition.error
                     results.solution_loader = SolSolutionLoader(None, None)
                 else:
-                    results.iteration_count = iters
-                    if ipopt_time_nofunc is not None:
+                    results.iteration_count = parsed_output_data['iters']
+                    parsed_output_data.pop('iters')
+                    if 'total_time' in parsed_output_data['cpu_seconds']:
+                        results.timing_info.total_seconds = parsed_output_data[
+                            'cpu_seconds'
+                        ]['total_time']
+                    if 'nofunc_time' in parsed_output_data['cpu_seconds']:
                         results.timing_info.ipopt_excluding_nlp_functions = (
-                            ipopt_time_nofunc
+                            parsed_output_data['cpu_seconds']['nofunc_time']
                         )
-
-                    if ipopt_time_func is not None:
-                        results.timing_info.nlp_function_evaluations = ipopt_time_func
-                    if ipopt_total_time is not None:
-                        results.timing_info.total_seconds = ipopt_total_time
+                        results.timing_info.nlp_function_evaluations = (
+                            parsed_output_data['cpu_seconds']['func_time']
+                        )
+                    parsed_output_data.pop('cpu_seconds')
+                    results.extra_info = parsed_output_data
+                    # Set iteration_log visibility to ADVANCED_OPTION because it's
+                    # a lot to print out with `display`
+                    results.extra_info.get("iteration_log")._visibility = (
+                        ADVANCED_OPTION
+                    )
```

Comment:
I'm confused. What type of object does `results.extra_info.get("iteration_log")` return? From below, I thought it would be a list.

Reply:
Correct - it's a list. (It can be a very _long_ list, because it captures all of the iterations. See https://github.com/IDAES/idaes-pse/issues/92#issuecomment-2828127621)

Reply:
I'm just confused about how it has the `_visibility` attribute...

Reply:
Oh, because it's an implicitly declared `ConfigValue`.